Memleak in crypto_openssl

From http://community.markus-raab.org:8080/job/elektra-clang/
```
&lt;b&gt;MPK&lt;/b&gt; ==3558== 16,384 bytes in 1 blocks are still reachable in loss record 572 of 572
==3558==    at 0x4C2AF2E: realloc (vg_replace_malloc.c:692)
==3558==    by 0x572742F: CRYPTO_realloc (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0)
==3558==    by 0x57B178B: lh_insert (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0)
==3558==    by 0x57B3E69: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0)
==3558==    by 0x57B383A: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0)
==3558==    by 0x6460589: ???
==3558==    by 0x64615A1: ???
==3558==    by 0x504E110: elektraPluginOpen (plugin.c:295)
==3558==    by 0x402138: test_init (test_internals.h:136)
==3558==    by 0x402138: main (testmod_crypto_openssl.c:21)
==3558== 
CYPTO        TESTS
==================


crypto_openssl
```

Do you need an extra issue for
```
src/plugins/crypto/openssl_operations.c: In function ‘elektraCryptoOpenSSLInit’:
src/plugins/crypto/openssl_operations.c:57:37: warning: unused parameter ‘errorKey’ [-Wunused-parameter]
 int elektraCryptoOpenSSLInit (Key * errorKey)
                                     ^
```
?

--- a/src/plugins/crypto/CMakeLists.txt
+++ b/src/plugins/crypto/CMakeLists.txt
@@ -182,7 +182,6 @@ add_plugin (crypto_botan
 if (ADDTESTING_PHASE)
 	if (HAS_OPENSSL_4SURE)
 		add_plugintest (crypto_openssl
-				MEMLEAK
 				LINK_PLUGIN "<no>")
 	endif ()
 
@@ -194,7 +193,6 @@ if (ADDTESTING_PHASE)
 
 	if (HAS_BOTAN_4SURE)
 		add_plugintest (crypto_botan
-				MEMLEAK
 				LINK_PLUGIN "<no>")
 	endif ()
 endif ()

--- a/src/plugins/crypto/openssl_operations.c
+++ b/src/plugins/crypto/openssl_operations.c
@@ -54,7 +54,7 @@ static Key * elektraCryptoReadParamIv (KeySet * config, Key * errorKey)
 	return iv;
 }
 
-int elektraCryptoOpenSSLInit (Key * errorKey)
+int elektraCryptoOpenSSLInit (Key * errorKey ELEKTRA_UNUSED)
 {
 	// initialize OpenSSL according to
 	// https://wiki.openssl.org/index.php/Library_Initialization

--- a/tests/valgrind.suppression
+++ b/tests/valgrind.suppression
@@ -1874,3 +1874,15 @@
    fun:test_config_errors
    fun:main
 }
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   match-leak-kinds: reachable
+   fun:realloc
+   fun:CRYPTO_realloc
+   fun:lh_insert
+   ...
+   fun:elektraPluginOpen
+   fun:test_init
+   fun:main
+}

--- a/tests/valgrind.suppression
+++ b/tests/valgrind.suppression
@@ -1824,3 +1824,53 @@
    fun:test_crypto_operations
    fun:main
 }
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   match-leak-kinds: indirect
+   fun:_Znwm
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   fun:_ZN5Botan10get_cipherERKSsNS_10Cipher_DirE
+   fun:_ZN5Botan10get_cipherERKSsRKNS_11OctetStringES4_NS_10Cipher_DirE
+   obj:*
+   obj:*
+   fun:test_config_errors_internal
+   fun:test_config_errors
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   match-leak-kinds: indirect
+   fun:_Znwm
+   fun:_ZNSt6vectorIPN5Botan6FilterESaIS2_EE14_M_fill_insertEN9__gnu_cxx17__normal_iteratorIPS2_S4_EEmRKS2_
+   fun:_ZN5Botan6FilterC1Ev
+   fun:_ZN5Botan14CBC_EncryptionC1EPNS_11BlockCipherEPNS_28BlockCipherModePaddingMethodE
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   fun:_ZN5Botan10get_cipherERKSsNS_10Cipher_DirE
+   fun:_ZN5Botan10get_cipherERKSsRKNS_11OctetStringES4_NS_10Cipher_DirE
+   obj:*
+   obj:*
+   fun:test_config_errors_internal
+   fun:test_config_errors
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   match-leak-kinds: indirect
+   fun:_Znwm
+   fun:_ZNK5Botan7AES_2565cloneEv
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   obj:/usr/lib/libbotan-1.10.so.0.8
+   fun:_ZN5Botan10get_cipherERKSsNS_10Cipher_DirE
+   fun:_ZN5Botan10get_cipherERKSsRKNS_11OctetStringES4_NS_10Cipher_DirE
+   obj:*
+   obj:*
+   fun:test_config_errors_internal
+   fun:test_config_errors
+   fun:main
+}

